include/dca/io/hdf5/hdf5_writer.hpp

```diff
@@ -54,58 +54,60 @@ class HDF5Writer {
   std::string get_path();
 
   template <typename arbitrary_struct_t>
-  static void to_file(arbitrary_struct_t& arbitrary_struct, std::string file_name);
+  static void to_file(const arbitrary_struct_t& arbitrary_struct, std::string file_name);
 
   template <typename scalar_type>
   void execute(std::string name, scalar_type value);
 
   template <typename scalar_type>
-  void execute(std::string name, std::pair<scalar_type, scalar_type>& value);
+  void execute(std::string name, const std::pair<scalar_type, scalar_type>& value);
```

Comment:
For correctness (and efficiency), I suspect that all those `std::string name` vars in the arguments should be `const std::string &name`. Might be a good idea to update them as well

Reply:
const ref and deep copy are functionally equivalent here, and given that is by far not performance critically, it is the same. But sure, I can mod it.